### PR TITLE
second commit

### PR DESCRIPTION
**Pull request by nebinvitusroy2002**
Added Exceptions and logs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -38,6 +38,12 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-security</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-api</artifactId>
```
Comment (aswin-mc-flycatch) [inline pom.xml:43]: we don't need this dependency , we need to use log from lombok 
ie, @Slf4j this annotation

File: src/main/java/com/spring/securitysignIn/service/serviceimpl/UserServiceImpl.java [hunk 2]
```diff
@@ -20,13 +25,26 @@ public UserServiceImpl(UserRepository userRepository){
     }
 
     @Override
-    public User findByUserName(String username){
-        return userRepository.findByUsername(username);
+    public Optional<User> findByUserName(String username){
+        logger.info("Searching for user with username: {}",username);
+        Optional<User> user = userRepository.findByUsername(username);
```
Comment (aswin-mc-flycatch) [inline src/main/java/com/spring/securitysignIn/service/serviceimpl/UserServiceImpl.java:30]: We need exception handling here. There is a possibility of an SQL exception occurring. Every repository call should handle exceptions appropriately

File: src/main/java/com/spring/securitysignIn/controller/UserController.java [hunk 3]
```diff
@@ -17,27 +18,37 @@
...
     public UserController(UserService userService){
         this.userService=userService;
     }
 
     @GetMapping("/home")
-    public UserDetails home(Principal principal){
-        return userDetailsService.loadUserByUsername(principal.getName());
+    public ResponseEntity<?> home(Principal principal){
+        try {
+            logger.info("Accessing /home for user: {}",principal.getName());
```
Comment (aswin-mc-flycatch) [inline src/main/java/com/spring/securitysignIn/controller/UserController.java:34]: logs to

Comment (nebinvitusroy2002) [reply src/main/java/com/spring/securitysignIn/controller/UserController.java:34]: So no need of log in UserController

**Pull request merged**